src/Handlers.jl

```diff
@@ -141,6 +141,16 @@ function handle(r::Router, req)
     return handle(handler, req)
 end
 
-handle(hf::HandlerFunction, http::HTTP.Stream) = handle_request(hf.func,http)
+function handle(r::Router, http::HTTP.Stream)
+    request::HTTP.Request = http.message
+    request.body = read(http)
+    hf = gethandler(r,request)
+    request.response::HTTP.Response = hf.func(request)
```

Comment:
This isn't quite right because it assumes the handler for a route is a `HandlerFunction`, whereas it could be any (even custom) Handler. More correct would be `... = handle(hf, request)`